starter-code/routes/auth.js

```diff
@@ -0,0 +1,106 @@
+require('dotenv').config()
+const express = require("express");
+const passport = require('passport');
+const { ensureLoggedIn, ensureLoggedOut } = require('connect-ensure-login');
+const multer = require('multer');
+const upload = multer({ dest: './public/uploads/' });
+const bodyParser = require('body-parser');
+
+
+
+const router = express.Router();
+
+router.get('/auth/facebook', passport.authenticate('facebook'));
+router.get('/auth/facebook/callback', passport.authenticate('facebook', { successRedirect: "/mytrips", failureRedirect: '/login' }), )
+
+
+router.get('/', (req, res, next) => {
+	res.render('index')
+})
+
+router.get('/mytrips', ensureLoggedIn('/'), (req, res, next) => {
+	const user = req.body.user
+	Trip.find({ userID: req.user._id },
+		(error, trips) => {
+			if (error) throw error
+			else res.render('mytrips', { user, trips })
+		})
+})
+
+auth.post('/mytrips/new', ensureLoggedIn('/'), upload.single('image'), (req, res, next) => {
+
+	const newTrip = new Trip({
+		userID: req.user._id,
+		displayName: req.user.displayName,
+		destination: req.body.destination,
+		description: req.body.description,
+		filename: req.body,
+		pic_path: `/uploads/${req.file.filename}`,
+		pic_name: req.file.originalname
+	})
+
+	newTrip.save((err) => {
+		res.render('mytrips', { newTrip });
+	});
+});
+
+router.get('/mytrips/new', ensureLoggedIn('/'), (req, res, next) => {
+	const user = req.body.user
+	res.render('new', { user })
+})
+
+router.get('/mytrips/edit/:trip_id', ensureLoggedIn('/'), (req, res, next) => {
+	Trip.find({ userID: req.params.user_id },
+		(error, trips) => {
+			if (error) throw error
+			else {
+				User.find({ _id: req.params.user_id },
```

Comment:
No param is called `user_id`. I believe you want to get the user from `req.user` instead.